drgn/helpers/linux/cgroup.py

```diff
@@ -0,0 +1,77 @@
+# SPDX-License-Identifier: GPL-3.0+
+
+"""
+Cgroup
+------
+
+The ``drgn.helpers.linux.cgroup`` module provides helpers for working with
+cgroup interface in :linux:`include/linux/cgroup.h`. Only cgroup v2 is
+supported.
+"""
+
+import os
+
+from drgn import Object, container_of
+
+
+__all__ = [
+        'cgroup_name',
+        'cgroup_parent',
+        'cgroup_path',
+        'sock_cgroup',
+]
+
+
+def sock_cgroup(skcd):
+    """
+    .. c:function:: struct cgroup *sock_cgroup(struct sock_cgroup_data *skcd)
+
+    Get cgroup of a socket by ``struct sock_cgroup_data`` member embedded to
+    every ``struct sock``.
+
+
+    :return: ``struct cgroup`` object.
+    """
+    return Object(skcd.prog_, 'struct cgroup', address=skcd.val)
+
+
+def cgroup_parent(cgrp):
+    """
+    .. c:function:: struct cgroup *cgroup_parent(struct cgroup *cgrp)
+
+    Get parent of cgroup if it exists.
+
+    :return: ``struct cgroup`` object or None.
+    """
+    parent_css = cgrp.self.parent
+    if parent_css:
+        return container_of(parent_css, 'struct cgroup', 'self')
+    return None
+
+
+def cgroup_name(cgrp):
```

Comment:
sysfs also uses kernfs, so it'd be worth it to add `kernfs_name` and `kernfs_path` helpers and use those for `cgroup_name` and `cgroup_path`. Would you mind doing that?

Reply:
`kernfs_name` is simple, I'll do it

`kernfs_path` is more involved so I'll do simplified version of it, that later can be extended if needed. By "simplified" I mean I won't implement the whole logic of `kernfs_path_from_node_locked` that is used by `kernfs_path` underneath but will handle only scenario when `from == NULL`, i.e. the path is built from starting root, not from a random `kn` in the fs. LMK if it's not a good idea for some reason.